### DISCUSSION TITLE
Welcome to TinyMqtt Discussions!

### DISCUSSION THREAD

**Discussion by hsaturn** [hsaturn/TinyMqtt#3] [General]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by simp16** [comment]
Hello, first of all thank you very much for your work! I don't know if this is the right place to ask but I set up your simple-broker example on esp32 with no problems and it works if I publish from a windows mqtt client, but I do not receive messages from my device running Tasmota, even though it shows on Tasmota console that it is connected to this mqtt broker. Also, my final goal is to send mqtt from my Tasmota device and the have esp32 "relay" the messages to a public broker, would this be possible with TinyMqtt? Thank you very much once again.

**Comment by hsaturn** [comment]
ouch !!! Sorry Simp16, I only view today this comment.
-> No this is not the right place to write this. You should have opened a issue.
Yes, TinyMqtt should be able to forward a msg from tasmota device to a public broker.
But this may not be exactly what you need.

What I guess from your question is

Tasmota device (home) -> TinyMqttBroker (home, another device (or not)) -> public MqttBroker.

In that case, any mqtt message is transmitted to the public Mqtt Broker through TinyMqttBroker.
But ... because TinyMqttBroker is connected to an external broker, it does rely on it.
I mean, the TinyMqttBroker will not forward the message to internal subscribers, because the MqttBroker connected to another broker does not consider itself as a broker, but only as a forward device.

If you were having this confuguration

Internal Mqtt Broker -> connected to an external broker
Tasmota device 1 -> MqttBroker 
Tasmota device 2 -> MqttBroker
Other Mqtt device -> MqttBroker

The MqttBroker cannot directly forward device 1 msg to device 2 for example. The right path is
device 1 -> MqttBroker -> external broker -> MqttBroker -> device 2

There can only exist one "master" broker. (Because of Mqtt architecture).
My multiple broker architecture allows to be more resilient to any problem.

For example in the device 1/device 2 / MqttBroker / external broker configuration
The main broker (thus the external broker) can die, then MqttBroker takes the decision to be 'master'.

-> This is not fully implemented.
